src/main/java/org/jenkins_ci/plugins/flexible_publish/FlexibleMatrixAggregator.java

```diff
@@ -96,11 +96,18 @@ public boolean endRun(MatrixRun run) throws InterruptedException,
      */
     @Override
     public boolean endBuild() throws InterruptedException, IOException {
+        boolean wholeResult = true;
         for (ConditionalMatrixAggregator cma: aggregatorList) {
-            if (!cma.endBuild()) {
-                return false;
+            try {
+                if (!cma.endBuild()) {
+                    listener.error(String.format("[flexible-publish] aggregation with %s failed", cma.toString()));
+                    wholeResult = false;
+                }
+            } catch (Exception e) {
+                e.printStackTrace(listener.error(String.format("[flexible-publish] aggregation with %s is aborted due to exception", cma.toString())));
+                wholeResult = false;
```

Comment:
Matrix-project doesn't handle exceptions.
I'll work for that in [JENKINS-27066](https://issues.jenkins-ci.org/browse/JENKINS-27066).